### ISSUE TITLE
Atari 5200 secondary trigger and player-independent keys

### ISSUE THREAD

**Issue by carmiker** [atari800/atari800#114]
I have realized there is no clear way to use the second trigger input on the Atari 5200 controller. The SDL frontend does not seem to have a setting for it. You can trigger it by setting INPUT_key_shift, but this is not player-independent. In kat5200, the triggers are labelled "top" and "bottom". I know of at least one game that requires the second trigger, Moon Patrol. Is there something I am missing here?

Similarly, there does not seem to be a clear way to differentiate the console keys (Start, Pause, Reset) or numerical keypad buttons between input devices. I don't know if this is actually necessary for console keys, since I see the Track Ball controller only has one set of those keys, but it does have two sets of numerical keys, and two sets of triggers.

**Comment by joysfera** [comment]
It would help a lot if we had a developer that would know the Atari 5200, would love its games and would want to have them supported properly in the Atari800 emulator.

**Comment by PRMan99** [comment]
Sure thing. How do I compile this on Windows?

**Comment by carmiker** [comment]
Since there has been no movement on this in 3 years, I am forced into the position of needing to become an expert on the Atari 5200. I will do the research and implementation required to support the system at a "Tier 1" level by writing an emulator from scratch and share all the documentation and license the code in such a way that it will also help the Atari800 project.

**Comment by carmiker** [comment]
@mikrosk yes, you are absolutely correct! However, I made the comment as a sort of joke and a way to ensure I really do become an expert on the system. I have done other systems in depth such as the ColecoVision, the Sega 8-bit family, and the Neo Geo, writing an emulator which serves as a living document that anyone can read and reference. So I will do the same for the Atari 5200 and everyone can benefit. I've been known to submit PRs to related projects as I develop my own emulators :).

**Comment by groessler** [comment]
Any news? I'd like to have better 5200 support to enhance the cc65 atari5200 target. Currently the numpads etc of the controllers aren't handled by the runtime library.

**Comment by carmiker** [comment]
@groessler I have too many projects on the go, but I have actually started work on this and hope to allocate more time to it this year.
